### PR TITLE
Noise2inverse test

### PR DESCRIPTION
**Pull request by cshoebridge**

### CODE REVIEW

File: LION/optimizers/LIONsolver.py [hunk 1]
```diff
@@ -141,35 +158,68 @@ def set_validation(
...
+
+        self.load_folder = load_folder
+        self.do_load_checkpoint = do_load
 
     def set_checkpointing(
         self,
         checkpoint_fname: str,
         checkpoint_freq: int = 10,
-        load_checkpoint: bool = False,
     ):
```
Comment (AnderBiguri) [inline LION/optimizers/LIONsolver.py:201]: new optional args:  save_path=None, load_path=None.

if checkpoint_fname is "full path" and save/load are None -> save_path/load_path are those
else read from this variables
else error.

File: LION/optimizers/LIONsolver.py [hunk 2]
```diff
@@ -141,35 +158,68 @@ def set_validation(
...
+    def set_loading(self, load_folder: str | pathlib.Path, do_load: bool = False):
+        if isinstance(load_folder, str):
+            load_folder = pathlib.Path(load_folder)
+        if not load_folder.is_dir():
+            raise ValueError(
+                f"Save folder '{load_folder}' is not a directory, failed to set saving."
+            )
+
+        self.load_folder = load_folder
+        self.do_load_checkpoint = do_load
```
Comment (AnderBiguri) [inline LION/optimizers/LIONsolver.py:195]: Inside set_checkpoint for that

File: LION/optimizers/LIONsolver.py [hunk 3]
```diff
@@ -141,35 +158,68 @@ def set_validation(
...
-        load_checkpoint: bool = False,
     ):
         """
         This function sets the checkpointing
         """
-        if not self.save_folder.is_dir():
-            raise ValueError(f"Save folder '{self.save_folder}' is not a directory")
-
+        if self.save_folder is None:
+            warnings.warn("Save folder not set. Please call set_saving")
```
Comment (AnderBiguri) [inline LION/optimizers/LIONsolver.py:206]: change

File: LION/optimizers/losses/LIONloss.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
+from abc import ABC, abstractmethod
```
Comment (AnderBiguri) [inline LION/optimizers/losses/LIONloss.py]: remove

File: LION/optimizers/losses/WGANloss.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import torch
+from LION.classical_algorithms.fdk import fdk
+from LION.models.LIONmodel import ModelInputType
+from LION.optimizers.losses.LIONloss import LIONtrainingLoss
+
+
+class WGANloss(LIONtrainingLoss):
```
Comment (AnderBiguri) [inline LION/optimizers/losses/WGANloss.py]: @AnderBiguri figure out if the name is correct

File: LION/optimizers/noise2inverse.py [hunk 6]
```diff
@@ -110,121 +110,89 @@ def mini_batch_step(self, sinos):
...
-        return test_loss
\ No newline at end of file
+
+        if was_training:
+            self.model.train()
+
+        return test_loss
+
+    # not convinced by this name
+    def process(self, sinos):
```
Comment (AnderBiguri) [inline LION/optimizers/noise2inverse.py]: call it reconstruct()

File: LION/optimizers/weaklysupervised_learning.py [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+from LION.optimizers.LIONsolver import LIONsolver, SolverParams
+
+# Parameter class
+from LION.utils.parameter import LIONParameter
+
+# standard imports
+from tqdm import tqdm
+
+
+class WeaklySupervisedSolver(LIONsolver):
```
Comment (AnderBiguri) [inline LION/optimizers/weaklysupervised_learning.py:21]: is this the right name?

File: LION/optimizers/weaklysupervised_learning.py [hunk 8]
```diff
@@ -0,0 +1,187 @@
+# numerical imports
+import torch
```
Comment (AnderBiguri) [inline LION/optimizers/weaklysupervised_learning.py:2]: remove

**Comment by cshoebridge** [comment]
remainder of issues will be resolved in the AR PR

**Comment by cshoebridge** [comment]
should be mergeable now 👍

**Pull request merged**